data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaTopicMetrics.java

```diff
@@ -0,0 +1,197 @@
+    private final Counter numberOfNegativeAcknowledgements;
+    private final Counter numberOfRecordsFailedToParse;
+    private final Counter numberOfDeserializationErrors;
+    private final Counter numberOfBufferSizeOverflows;
+    private final Counter numberOfPollAuthErrors;
+    private final Counter numberOfRecordsCommitted;
+    private final Counter numberOfRecordsConsumed;
+    private final Counter numberOfBytesConsumed;
+    private final Counter numberOfNonConsumers;
+
+    public KafkaTopicMetrics(final String topicName, final PluginMetrics pluginMetrics) {
+        this.pluginMetrics = pluginMetrics;
+        this.topicName = topicName;
+        this.updateTime = Instant.now().getEpochSecond();
+        this.metricValues = new HashMap<>();
+        initializeMetricNamesMap();
+        this.numberOfRecordsConsumed = pluginMetrics.counter(getTopicMetricName(NUMBER_OF_RECORDS_CONSUMED));
+        this.numberOfBytesConsumed = pluginMetrics.counter(getTopicMetricName(NUMBER_OF_BYTES_CONSUMED));
+        this.numberOfRecordsCommitted = pluginMetrics.counter(getTopicMetricName(NUMBER_OF_RECORDS_COMMITTED));
+        this.numberOfRecordsFailedToParse = pluginMetrics.counter(getTopicMetricName(NUMBER_OF_RECORDS_FAILED_TO_PARSE));
+        this.numberOfDeserializationErrors = pluginMetrics.counter(getTopicMetricName(NUMBER_OF_DESERIALIZATION_ERRORS));
+        this.numberOfBufferSizeOverflows = pluginMetrics.counter(getTopicMetricName(NUMBER_OF_BUFFER_SIZE_OVERFLOWS));
+        this.numberOfPollAuthErrors = pluginMetrics.counter(getTopicMetricName(NUMBER_OF_POLL_AUTH_ERRORS));
+        this.numberOfNonConsumers = pluginMetrics.counter(getTopicMetricName(NUMBER_OF_NON_CONSUMERS));
+        this.numberOfPositiveAcknowledgements = pluginMetrics.counter(getTopicMetricName(NUMBER_OF_POSITIVE_ACKNOWLEDGEMENTS));
+        this.numberOfNegativeAcknowledgements = pluginMetrics.counter(getTopicMetricName(NUMBER_OF_NEGATIVE_ACKNOWLEDGEMENTS));
+    }
+
+    private void initializeMetricNamesMap() {
+        this.metricsNameMap = new HashMap<>();
+        metricsNameMap.put("bytes-consumed-total", "bytesConsumedTotal");
+        metricsNameMap.put("records-consumed-total", "recordsConsumedTotal");
+        metricsNameMap.put("bytes-consumed-rate", "bytesConsumedRate");
+        metricsNameMap.put("records-consumed-rate", "recordsConsumedRate");
+        metricsNameMap.put("records-lag-max", "recordsLagMax");
+        metricsNameMap.put("records-lead-min", "recordsLeadMin");
+        metricsNameMap.put("commit-rate", "commitRate");
+        metricsNameMap.put("join-rate", "joinRate");
+        metricsNameMap.put("incoming-byte-rate", "incomingByteRate");
+        metricsNameMap.put("outgoing-byte-rate", "outgoingByteRate");
+        metricsNameMap.forEach((metricName, camelCaseName) -> {
+            if (!metricName.contains("-total")) {
+                pluginMetrics.gauge(getTopicMetricName(camelCaseName), metricValues, metricValues -> {
```

Comment:
This code should be unit tested.

Reply:
Also, it might make sense to invert the if-else with the gauge to clarify the behavior some. The condition is really on the metric function, not inside it.

Something like:

```
if(metricName.equals("records-lag-max") {
  pluginMetrics.gauge(getTopicMetricName(camelCaseName), metricValues, metricValues -> {
    double max = 0.0;
    for (Map.Entry<KafkaConsumer, Map<String, Double>> entry : metricValues.entrySet()) {
    if (entry.getValue().get(metricName) > max) {
        max = entry.getValue().get(metricName);
      }
    }
    return max;
  });
else if(...)
  pluginMetrics.gauge(...)
}
```